src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -23,28 +26,76 @@
 @WebServlet("/auth")
 public class AuthServlet extends HttpServlet {
 
-  // Boolean that holds the current sign in status.
-  private boolean signedIn;
-
   @Override
-  public void init() {
-    signedIn = false;
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    // Set up user auth objects.
+    UserService userService = UserServiceFactory.getUserService();
+    UserAuth userAuth;
+
+    // Set redirect URL after login / logout as index page.
+    final String redirectUrl = "/";
```

Comment:
optional: I'll second Adams comment I think this could reasonably be a class constant since it's always the same regardless of the context in which this method was called.

Reply:
Done!

Reply:
Actually, I still have to add static; changing this now.

Reply:
Done!